backend/src/schema/resolvers/reports.js

```diff
@@ -76,14 +66,24 @@ export default {
           filterClause = ''
       }
 
-      if (params.closed) filterClause = 'AND report.closed = true'
+      switch (params.closed) {
+        case true:
+          filterClause = 'AND report.closed = true'
+          break
+        case false:
+          filterClause = 'AND report.closed = false'
+          break
+        default:
+          break
+      }
 
       const offset =
         params.offset && typeof params.offset === 'number' ? `SKIP ${params.offset}` : ''
       const limit = params.first && typeof params.first === 'number' ? `LIMIT ${params.first}` : ''
 
-      const reportReadTxPromise = session.readTransaction(async transaction => {
-        const allReportsTransactionResponse = await transaction.run(
+      const reportsReadTxPromise = session.readTransaction(async transaction => {
+        const reportsTransactionResponse = await transaction.run(
+          // !!! this Cypher query returns multiple reports on the same resource! i will create an issue for refactoring (bug fixing)
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-20T10:27:34Z" title="Thursday, February 20th 2020, 11:27:34 am +01:00">Feb 20, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -76,14 +66,24 @@ export default {
           filterClause = ''
       }
 
-      if (params.closed) filterClause = 'AND report.closed = true'
+      switch (params.closed) {
+        case true:
+          filterClause = 'AND report.closed = true'
+          break
+        case false:
+          filterClause = 'AND report.closed = false'
+          break
+        default:
+          break
+      }
 
       const offset =
         params.offset && typeof params.offset === 'number' ? `SKIP ${params.offset}` : ''
       const limit = params.first && typeof params.first === 'number' ? `LIMIT ${params.first}` : ''
 
-      const reportReadTxPromise = session.readTransaction(async transaction => {
-        const allReportsTransactionResponse = await transaction.run(
+      const reportsReadTxPromise = session.readTransaction(async transaction => {
+        const reportsTransactionResponse = await transaction.run(
+          // !!! this Cypher query returns multiple reports on the same resource! i will create an issue for refactoring (bug fixing)
```


```suggestion
         
```
it's intended to return a list of all reports, even if there are two that point to the same resource.